jenkins-master/jobs/mozilla-aurora_functional/config.xml

```diff
@@ -48,32 +48,9 @@
       <contributors/>
     </EnvInjectJobProperty>
   </properties>
-  <scm class="hudson.plugins.git.GitSCM" plugin="git@2.3.5">
-    <configVersion>2</configVersion>
-    <userRemoteConfigs>
-      <hudson.plugins.git.UserRemoteConfig>
-        <url>https://github.com/mozilla/firefox-ui-tests.git</url>
-      </hudson.plugins.git.UserRemoteConfig>
-    </userRemoteConfigs>
-    <branches>
-      <hudson.plugins.git.BranchSpec>
-        <name>*/mozilla-aurora</name>
-      </hudson.plugins.git.BranchSpec>
-    </branches>
-    <doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations>
-    <submoduleCfg class="list"/>
-    <extensions>
-      <hudson.plugins.git.extensions.impl.LocalBranch>
-        <localBranch>mozilla-aurora</localBranch>
-      </hudson.plugins.git.extensions.impl.LocalBranch>
-      <hudson.plugins.git.extensions.impl.CloneOption>
-        <shallow>true</shallow>
-        <reference></reference>
-      </hudson.plugins.git.extensions.impl.CloneOption>
-      <hudson.plugins.git.extensions.impl.RelativeTargetDirectory>
-        <relativeTargetDir>firefox-ui-tests</relativeTargetDir>
-      </hudson.plugins.git.extensions.impl.RelativeTargetDirectory>
-    </extensions>
```

Comment:
Yes, the mozharness action deals with submodules. You need to configure the
revision of firefox-medie-tests and the revision of the firefox-ui-tests
submodule. Here's the code:
https://dxr.mozilla.org/mozilla-central/source/testing/mozharness/mozharness/mozilla/testing/firefox_media_tests.py#201

In any case, the submodule will go away once firefox_puppeteer is a
separate package in-tree.

Maja Frydrychowicz
irc: maja_zf

On Thu, Nov 19, 2015 at 2:38 PM, Syd Polk <notifications@github.com> wrote:

> In jenkins-master/jobs/mozilla-aurora_functional/config.xml
> <https://github.com/mozilla/mozmill-ci/pull/698#discussion_r45389325>:
>
> > -      </hudson.plugins.git.BranchSpec>
> > -    </branches>
> > -    <doGenerateSubmoduleConfigurations>false</doGenerateSubmoduleConfigurations>
> > -    <submoduleCfg class="list"/>
> > -    <extensions>
> > -      <hudson.plugins.git.extensions.impl.LocalBranch>
> > -        <localBranch>mozilla-aurora</localBranch>
> > -      </hudson.plugins.git.extensions.impl.LocalBranch>
> > -      <hudson.plugins.git.extensions.impl.CloneOption>
> > -        <shallow>true</shallow>
> > -        <reference></reference>
> > -      </hudson.plugins.git.extensions.impl.CloneOption>
> > -      <hudson.plugins.git.extensions.impl.RelativeTargetDirectory>
> > -        <relativeTargetDir>firefox-ui-tests</relativeTargetDir>
> > -      </hudson.plugins.git.extensions.impl.RelativeTargetDirectory>
> > -    </extensions>
>
> Right now, firefox-media-tests has firefox-ui-tests as a submodule. Will
> the mozharness action which calls get deal with submodules? Or is there
> some other mechanism for insuring that firefox-media-tests can find
> firefox-ui-tests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/mozmill-ci/pull/698/files#r45389325>.
>